.palantir/revapi.yml

```diff
@@ -1456,6 +1456,41 @@ acceptedBreaks:
+      - code: "java.method.visibilityReduced"
+        old: "method B org.apache.iceberg.rest.responses.BaseScanTaskResponse.Builder<B\
+          \ extends org.apache.iceberg.rest.responses.BaseScanTaskResponse.Builder<B\
+          \ extends org.apache.iceberg.rest.responses.BaseScanTaskResponse.Builder<B,\
+          \ R extends org.apache.iceberg.rest.responses.BaseScanTaskResponse>, R>, R\
+          \ extends org.apache.iceberg.rest.responses.BaseScanTaskResponse>::withSpecsById(java.util.Map<java.lang.Integer,\
+          \ org.apache.iceberg.PartitionSpec>)"
+        new: "method B org.apache.iceberg.rest.responses.BaseScanTaskResponse.Builder<B\
+          \ extends org.apache.iceberg.rest.responses.BaseScanTaskResponse.Builder<B\
+          \ extends org.apache.iceberg.rest.responses.BaseScanTaskResponse.Builder<B,\
+          \ R extends org.apache.iceberg.rest.responses.BaseScanTaskResponse>, R>, R\
+          \ extends org.apache.iceberg.rest.responses.BaseScanTaskResponse>::withSpecsById(java.util.Map<java.lang.Integer,\
+          \ org.apache.iceberg.PartitionSpec>)"
+        justification: "specById API is not part of Plan API response spec. No actual\
+          \ breakage because it never worked before."
+      - code: "java.method.visibilityReduced"
+        old: "method java.util.Map<java.lang.Integer, org.apache.iceberg.PartitionSpec>\
+          \ org.apache.iceberg.rest.responses.BaseScanTaskResponse.Builder<B extends\
+          \ org.apache.iceberg.rest.responses.BaseScanTaskResponse.Builder<B extends\
+          \ org.apache.iceberg.rest.responses.BaseScanTaskResponse.Builder<B, R extends\
+          \ org.apache.iceberg.rest.responses.BaseScanTaskResponse>, R>, R extends org.apache.iceberg.rest.responses.BaseScanTaskResponse>::specsById()"
+        new: "method java.util.Map<java.lang.Integer, org.apache.iceberg.PartitionSpec>\
+          \ org.apache.iceberg.rest.responses.BaseScanTaskResponse.Builder<B extends\
+          \ org.apache.iceberg.rest.responses.BaseScanTaskResponse.Builder<B extends\
+          \ org.apache.iceberg.rest.responses.BaseScanTaskResponse.Builder<B, R extends\
+          \ org.apache.iceberg.rest.responses.BaseScanTaskResponse>, R>, R extends org.apache.iceberg.rest.responses.BaseScanTaskResponse>::specsById()"
+        justification: "specById API is not part of Plan API response spec. No actual\
+          \ breakage because it never worked before."
+      - code: "java.method.visibilityReduced"
+        old: "method java.util.Map<java.lang.Integer, org.apache.iceberg.PartitionSpec>\
+          \ org.apache.iceberg.rest.responses.BaseScanTaskResponse::specsById()"
+        new: "method java.util.Map<java.lang.Integer, org.apache.iceberg.PartitionSpec>\
+          \ org.apache.iceberg.rest.responses.BaseScanTaskResponse::specsById()"
+        justification: "specById API is not part of Plan API response spec. No actual\
+          \ breakage because it never worked before."
```

Comment:
the `because it never worked before` seems rather confusing here. Maybe just omit this part?

Reply:
> Maybe just omit this part?

Agree, i just wanted to emphasize that there is no known implementation who would be using it. but it may be possible that some java based catalog have started implementing the server side planning with these models but even for them it would be like a no-op, as the client would not be relying on the field they populated nor it will show up in the rest response (or post-serde)

Thinking more about marking this `@deprecated`, since we are still using its gonna start generating warnings, which IMHO is fine

<img width="628" height="233" alt="Screenshot 2025-11-03 at 6 32 00 AM" src="https://github.com/user-attachments/assets/e7a91762-1c36-430c-9b78-759da80a6136" />

please let me know your thoughts considering above

Reply:
I think having the warning is fine for one release. After one release we'll reduce visibility and remove the deprecation - which will also remove the warning again